src/ts/order.ts

```diff
@@ -176,12 +176,16 @@ export type NormalizedOrder = Omit<Order, "validTo" | "appData" | "kind"> & {
  * @returns A 32-byte hash encoded as a hex-string.
  */
 export function normalizeOrder(order: Order): NormalizedOrder {
-  return {
+  const normalizedOrder = {
     receiver: ethers.constants.AddressZero,
     ...order,
     validTo: timestamp(order.validTo),
     appData: hashify(order.appData),
   };
+  if (normalizedOrder.receiver === undefined) {
```

Comment:
nanonit: It's a bit weird to explicitly set receiver to AddressZero above and then check again (I assume it can be overridden by order?). Maybe add a comment or remove line 180?